2018/001-external-authentication.md

```diff
@@ -0,0 +1,126 @@
+# External Authentication Integration
+
+| Field  | Value      |
+|:-------|:-----------|
+| Status | Draft      |
+| Date   | 2018-05-03 |
+
+## Introduction
+
+### Problem description
+
+Many enterprises will have a pre-existing authentication service such as LDAP,
+SAML 2.0, etc and wish to use it to handle authentication of users for
+Kubernetes.
+
+### Proposed change
+
+* Allow operators to provide connection details for external authentication
+  systems via Velum.
```

Comment:
I asked a couple of customers about this and they expressed they'd feel a lot more confident with it as a field in velum rather than SSHing to the admin node (which they're trying to treat less like a "pet" - especially once the admin node containers run on other nodes which I believe is the pan for v4/v5). They want the field to have a "test" button before committing so the details can be validated (they understand most of velum at the moment does not validate fields but authentication testing would be one of the ones they'd like validated). The field should also be editable post-bootstrap if their authentication server changes (as it's likely they'd start off connecting to a test auth infra before letting it access their prod one but wouldn't want to rebuild the cluster).

Reply:
You could extend the public cloud 'first-run' script: https://github.com/SUSE/pubcloud/tree/master/generic/src/CaaSP/caasp-admin-setup